day3b.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+from dataclasses import dataclass
+import re
+from collections import defaultdict
+
+NUMBER_PATERN = re.compile(r"\d+")
+
+with open("input3a.txt", "r") as fp:
+    lines = fp.read().strip().splitlines()
+
+@dataclass
+class PartNumber:
+    row: int
+    start: int
+    end: int
+    value: int
+
+
+def find_maybe_part_numbers(lines: list[str]) -> list[PartNumber]:
+    nums = []
+    for row, line in enumerate(lines):
+        for match in NUMBER_PATERN.finditer(line):
+            nums.append(PartNumber(row, match.start(), match.end(), int(match.group())))
+    return nums
```

Comment:
We should name `nums` -> `part_numbers`:
```suggestion
    part_numbers = []
    for row, line in enumerate(lines):
        for match in NUMBER_PATERN.finditer(line):
            part_numbers.append(PartNumber(row, match.start(), match.end(), int(match.group())))
    return part_numbers
```